cache/src/main/java/io/envoyproxy/controlplane/cache/CacheStatusInfo.java

```diff
@@ -20,14 +16,7 @@
 
   private final T nodeGroup;
 
-  @GuardedBy("lock")
-  private final Map<Long, Watch> watches = new HashMap<>();
-
-  private final ReadWriteLock lock = new ReentrantReadWriteLock();
-  private final Lock readLock = lock.readLock();
-  private final Lock writeLock = lock.writeLock();
-
-  @GuardedBy("lock")
+  private final ConcurrentMap<Long, Watch> watches = new ConcurrentHashMap<>();
   private long lastWatchRequestTime;
```

Comment:
Since it's not guarded by lock anymore. Maybe add a comment that it's possible to get stale values here? Especially that the class is marked `@ThreadSafe`

or maybe add `volatile` or `AtomicLong`?

Reply:
`volatile` could work, but even so, i think there's actually little difference in the guarantees being provided, with the lock in place, we only guaranteed that no other could be writing `lastWatchRequestTime` as we read it, but it could actually be changed immediately after we return, so we could actually be reading sort of a stale value.

Of course `volatile`would help to get more up-to-date values